src/common/preservedTableState.component.tsx

```diff
@@ -367,6 +374,25 @@ export const usePreservedTableState = (props?: UsePreservedTableStateProps) => {
     [defaultState.cF, updateSearchParams]
   );
 
+  const setColumnFilterFns = useCallback(
+    (updaterOrValue: Updater<MRT_ColumnFilterFnsState>) => {
+      updateSearchParams((prevState: StatePartial) => {
+        const newValue = getValueFromUpdater(
+          updaterOrValue,
+          prevState.cFn || defaultState.cFn
+        );
+        const initialValue = defaultState.cFn;
+        const isDefaultState =
+          JSON.stringify(initialValue) === JSON.stringify(newValue);
+        return {
+          ...prevState,
+          cFn: isDefaultState ? undefined : newValue,
+        };
+      });
+    },
+    [defaultState.cFn, updateSearchParams]
+  );
```

Comment:
Could the unit tests be updated to cover this as well? See `onColumnFiltersChange updates the state and url correctly` and similar.